### PR TITLE
Added Diag Setting to push AuditEvents to LA.

### PR DESCRIPTION
**Pull request by TomKimber**
Resolves: PTASK5000229

Notes:
- Added Diag Setting to push AuditEvents to LA.

### CODE REVIEW

**Review by cakeben — APPROVED** [review]

**Review by timja — CHANGES_REQUESTED** [review]
not compatible with ithc, perftest and SDS as it.

discussing via slack

File: diagnostic-settings.tf [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+
+    retention_policy {
+      enabled = true
+      days    = 14
+    }
+  }
+}
+
+module "log_analytics_workspace" {
+  count = var.enable_diagnostic_setting ? 1 : 0
```
Comment (timja) [inline diagnostic-settings.tf:19]: I would remove this we don't generally flag this sort of thing, especially for key vaults, we can add an opt out if required later

**Review by timja — DISMISSED** [review]
thinking more about it i see why it works, similar to how private dns zone linking works with private endpoints

File: diagnostic-settings.tf (hunk 1 above)
Comment (TomKimber) [reply diagnostic-settings.tf]: Removed diagnostic setting toggle as suggested.

**Review by timja — APPROVED** [review]

**Pull request merged**